sherpa/models/tests/test_model.py

```diff
@@ -389,6 +393,86 @@ def test_get_mins_maxes_warns(caplog):
     assert msgs[5] == 'value of parameter box1d.xlow maximum is below hard minimum; setting to hard minimum'
 
 
+def test_set_hard_limit_warns(caplog):
+    """Setting the thawed pars is different to changing the par directly.
+
+    Check one of these cases. Created to address #1688.  Unfortunately
+    the check using caplog captures the logging output correctly, and
+    not as a used would see it (where the %s was not being replaced),
+    which means this can not check #1688 is fixed. DJB does not know
+    how to do such a check witout adding extensive testing machinery
+    which is not warranted in this case.
+
+    """
+
+    mdl = Sin("m")
+
+    # Check the defaults haven't changed.
+    #
+    assert mdl.thawedpars == pytest.approx([1, 0, 1])
+
+    # Check minimum values.
+    #
+    with caplog.at_level(logging.INFO, logger='sherpa'):
+        mdl.thawedpars = [0, -1, -1]
+
+    assert mdl.thawedpars == pytest.approx([1e-10, 0, 1e-5])
+    assert len(caplog.records) == 3
+
+    # Check maximum values.
+    #
+    big = 4e40
+    with caplog.at_level(logging.INFO, logger='sherpa'):
+        mdl.thawedpars = [big, big, big]
+
+    assert mdl.thawedpars == pytest.approx([10, hugeval, hugeval])
+    assert len(caplog.records) == 6
+
+    msgs = []
+    for (log_name, log_level, message) in caplog.record_tuples:
+        assert log_level == logging.WARNING
+        assert log_name == 'sherpa.models.model'
+        msgs.append(message)
+
+    def minstr(name):
+        return f"value of parameter {name} is below minimum; setting to minimum"
+
+    def maxstr(name):
+        return f"value of parameter {name} is above maximum; setting to maximum"
+
+    assert msgs[0] == minstr("m.period")
+    assert msgs[1] == minstr("m.offset")
+    assert msgs[2] == minstr("m.ampl")
+    assert msgs[3] == maxstr("m.period")
+    assert msgs[4] == maxstr("m.offset")
+    assert msgs[5] == maxstr("m.ampl")
+
+
+def test_set_pars_errors_soft_limits():
+    """Do we get errors about soft limits
+
+    Similar to test_get_mins_maxes_warns but it's an error,
+    not a warning.
```

Comment:
I strongly support opening a new issue about this because this is old behavior. I can imagine that there is a subtle reason to say why, but perhaps it is just a change that happened accidentally.

Reply:
See #1700